pkg/whip/whip_handler.go

```diff
@@ -238,6 +238,21 @@ func (h *whipHandler) getSDPAnswer(ctx context.Context, offer *webrtc.SessionDes
 		return "", psrpc.NewErrorf(psrpc.DeadlineExceeded, "timed out while waiting for ICE candidate gathering")
 	}
 
+	parsedAnswer, err := h.pc.LocalDescription().Unmarshal()
+	if err != nil {
+		return "", err
+	}
+	if len(parsedAnswer.MediaDescriptions) != h.expectedTrackCount {
+		return "", errors.ErrUnsupportedDecodeFormat
+	}
+	for _, m := range parsedAnswer.MediaDescriptions {
```

Comment:
This loop is not ideal, but I can't think of another way to identify that a codec is not supported with the current Pion semantics, short of validating the offer manually.